bindings_wasm/src/local_storage_persistence.rs

```diff
@@ -1,5 +1,17 @@
+use base64::{engine::general_purpose, Engine as _};
+use thiserror::Error;
+use wasm_bindgen::JsValue;
 use xmtp::persistence::Persistence;
 
+#[derive(Error, Debug)]
+pub enum LocalStoragePersistenceError {
+    #[error("Failed to read/write from local storage")]
+    ReadWriteError { native_error: JsValue },
+
+    #[error("Failed to deserialize from local storage")]
+    DeserializationError(#[from] base64::DecodeError),
+}
+
```

Comment:
Like the idea of using this enum structure. Very clear about how a module can fail.

From a handling POV is there a mechanism to inform developers which Errors are possible for a given method/function? 

Looking at `fn write(&mut self, key: &str, value: &[u8]) -> Result<(), Self::Error>`

the handler code would look like:
```rust
match writer.write("KEY", "bytes".as_bytes()) {
  Ok(v) => success(v),
  Err(LocalStoragePersistenceError::ReadWriteError) => retry(),
  Err(LocalStoragePersistenceError::DeserializationError) => notPossible!(),
}
```

It's not immediately clear which functions return which errors without looking at the source. Worse if developers ignore part of the pattern matching:

```rust
match writer.write("KEY", "bytes".as_bytes()) {
  Ok(v) => success(v),
  Err(LocalStoragePersistenceError::ReadWriteError) => retry(),
  Err(_) => Ignore(),
}
```

Then potentially uncaught errors will only be discoverable at runtime. Any guidance on how these errors should be handled from the calling perspective?